packages/aws-cdk-lib/aws-cloudwatch-actions/lib/lambda.ts

```diff
@@ -22,13 +22,17 @@ export class LambdaAction implements cloudwatch.IAlarmAction {
    * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricAlarm.html
    */
   bind(scope: Construct, alarm: cloudwatch.IAlarm): cloudwatch.AlarmActionConfig {
-    const idPrefix = FeatureFlags.of(scope).isEnabled(LAMBDA_PERMISSION_LOGICAL_ID_FOR_LAMBDA_ACTION) ? alarm.node.id : '';
-    this.lambdaFunction.addPermission(`${idPrefix}AlarmPermission`, {
-      sourceAccount: Stack.of(scope).account,
-      action: 'lambda:InvokeFunction',
-      sourceArn: alarm.alarmArn,
-      principal: new iam.ServicePrincipal('lambda.alarms.cloudwatch.amazonaws.com'),
-    });
+    const flag = FeatureFlags.of(scope).isEnabled(LAMBDA_PERMISSION_LOGICAL_ID_FOR_LAMBDA_ACTION);
+    const idPrefix = flag ? alarm.node.id : '';
+    const permissionId = `${idPrefix}AlarmPermission`;
+    if (!this.lambdaFunction.permissionsNode.tryFindChild(permissionId) || !flag) {
```

Comment:
Not sure I follow here 🤔

Why is the `|| !flag` necessary? 
From the feature flag definition:

> When enabled, the logical ID of a Lambda permission for a Lambda action includes an alarm ID.

Ie, it should just control how we define `permissionId`, not the flow of the function.

Also, how is this going to fix the mentioned bug?
With `if (!this.lambdaFunction.permissionsNode.tryFindChild(permissionId)` we're just ignoring the IDs that would cause an exception.
We wouldn't need a feature flag in that case, but we would simply skip adding lambda permissions without feedback to the user (with consequent potential issues).

Reply:
Thanks for the review! I reverted the feature flag to only affect the logical ID and added a comparison with the permission source ARN and alarm ARN.

Reply:
Thanks for the quick follow-up!
Now I get why you need the check on the feature flag:

> it breaks the throws when multiple alarms are created for the same lambda if feature flag is set to false test because it no longer throws.

I wonder if this fix is necessary. 
In what scenarios is useful to specify the same Lamdba for different actions on the same alarm? 🤔

Reply:
The use case that stumbled me into this bug 😅 was I wanted to send a POST request from a lambda to a Slack webhook that would send a message in an alerts channel. I wanted to have updates on alert and ok to have proper observability.

Reply:
I see. 
Just to understand the context fully:
Why use the same Lambda for different actions from the same alarm? 
Isn't it better to define a function for each action? (I guess you're probably defining the logic for handling the different actions in the function's code now)
My point is that we shouldn't fix the implementation to allow users to define non-optimal implementations.
But maybe I'm missing something, please let me know if that is the case.

Reply:
I see your point. However, for my use case the only difference I have between the alarm and ok actions is the emoji that I send in the message lol. My thought is it really necessary to have two lambdas when it can be expressed as a ternary?

Additionally, should the CDK be a blocker for architectural decisions?